### PR TITLE
Remove unnecessary guild redownloading (#383)

### LINKED ISSUE #383: DiscordApiClient current user guilds improperly queries individual guilds for non-DiscordClient implementations, leading to 401s
# Summary
The `DiscordApiClient` improperly handles response from Discord's API when querying `/users/@me/guilds`.

# Details
Per [implementation of GetCurrentUserGuildsAsync](https://github.com/DSharpPlus/DSharpPlus/blob/master/DSharpPlus/Net/DiscordApiClient.cs#L1004-L1021), the DiscordApiClient attempts to query the API for each individual guild received (possibly to fill in the blanks). This breaks DiscordRestClient, as it will attempt to perform an operation which the client might not be authorized to do.

The clients who call the method are immediately greeted with a 401.

# Steps to reproduce
1. Create an instance of `DiscordRestClient` with valid token
2. Call `GetCurrentUserGuildsAsync()`
3. You will get a 401

# Notes
The implementation should probably be unified, or someone should figure out why it was implemented like this in the first place.

### PR DESCRIPTION
**Pull request by js6pak**
# Summary
Fixes #383 

# Details
Repair #383, while not breaking #275, removing unnecessary guild downloads that only slow down the entire request. Even Kiritsu doesn't know why he wrote it now.

### CODE REVIEW

File: DSharpPlus/Entities/DiscordUser.cs [hunk 1]
```diff
@@ -84,6 +85,12 @@ public string DefaultAvatarUrl
         [JsonProperty("email", NullValueHandling = NullValueHandling.Ignore)]
         public virtual string Email { get; internal set; }
 
+        /// <summary>
+        /// Gets the user's email address.
```
Comment (Emzi0767) [inline DSharpPlus/Entities/DiscordUser.cs]: If you're doing that, and generally you shouldn't mix changes within PRs, please fix the docstring.

**Comment by js6pak** [comment]
Can you check cf69e7a? Can bot have `Bearer` token type?

**Comment by Emzi0767** [comment]
The brace in the enumeration, put it on new line.

As far as the bot thing goes: theoretically bearer is an application that acts on user's behalf, so application calls are valid. I'm not sure however.

**Comment by js6pak** [comment]
Okay, but `https://discordapp.com/api/v7/oauth2/applications/@me` need `bot` scope

**Comment by Emzi0767** [comment]
You sure?

**Comment by js6pak** [comment]
I'm think yes
image
image

**Comment by Emzi0767** [comment]
A'ight

**Pull request merged**